src/modality-specific-files/eye-tracking.md

```diff
@@ -0,0 +1,213 @@
+# Eye-tracking including gaze position and pupil size
+
+Support for eye-tracking dataset developed as a
+[BIDS Extension Proposal](../extensions.md#bids-extension-proposals).
+Please see [Citing BIDS](../introduction.md#citing-bids) on how to
+appropriately credit this extension when referring to it in the context
+of the academic literature.
+
+## Terminology and conventions
+
+Eyetracking-BIDS is fully consistent with the BIDS specification as a whole.
+However, BIDS was initially developed in the context of MRI,
+so some terminology may be unfamiliar to researchers from other fields.
+This section adds clarifications to
+[Common Principles - Definitions](../common-principles.md#definitions) for the
+eye-tracking context.
+
+-   **Eye-tracker** - refers to the apparatus allowing the records of gaze
+    position and/or pupil size.
+
+-   **Screen** - Video display used to present visual stimulation (for example
+    projector, monitor, tablet).
+
+## Eye-tracking data
+
+With EyeTracking-BIDS, we wish to promote the adoption of good practices in
+the management of scientific data.
+
+### Template
+
+<!--
+This block generates a filetree exanple.
+A guide for editing it can be found at
+ https://github.com/bids-standard/bids-specification/blob/master/macros_doc.md.
+-->
+{{ MACROS___make_filetree_example(
+
+   {
+   "sub-<label>": {
+      "<datatype>": {
+         "<matches>_eyetrack.json": "",
+         "<matches>_eyetrack.tsv[.gz]": "",
+         },
+      }
+   }
+
+) }}
+
+Eye-tracking data MUST be stored in the main data recording modality or
+ `<datatype>` directory (for example `func` when combined with fMRI, or
+ `beh` when combined with behavioral measures).
+
+In the template filenames, the `<matches>` part corresponds to task filename
+before the suffix. For example, when fMRI is considered the main modality with
+files such as `sub-01_ses-1_task-pRF_run-1_bold.nii.gz`, `<matches>` would
+then correspond to `sub-01_ses-1_task-pRF_run-1`.
+
+### Tabular file (`*_eyetrack.tsv[.gz]`)
+
+The eye-tracking data must be saved in `.tsv` or `.tsv.gz` format.
+The tabular files consist of one row per sample and a set of REQUIRED and
+OPTIONAL columns.
+
+<!--
+This block generates a columns table.
+The definitions of these fields can be found in
+  src/schema/rules/tabular_data/eyetrack.yaml
+and a guide for using macros can be found at
+ https://github.com/bids-standard/bids-specification/blob/master/macros_doc.md
+-->
+{{ MACROS___make_columns_table("eyetrack.EyeTracking") }}
+
+Throughout BIDS you can indicate missing values with `n/a` (for "not
+available").
+
+We encourage users to put the eye-tracking raw data within the
+`/sourcedata` directory (for example put eye-tracker proprietary files before
+conversion to `tsv.gz` format in the sourcedata directory).
```

Comment:
If this is something the BIDS community wants, it should be done elsewhere (covering all the modalities that are included in this *encouragement*).

If the BIDS steering group wants to recommend this, there are mechanisms for BIDS to adopt changes. Sneaking this into a random BEP doesn't seem to be an appropriate interpretation of the bylaws.

Finally, the BIDS specifications aim to be very explicit. If this is not shared widely to become a "RECOMMENDED" practice (to which I would strongly oppose), then the specs can, at most remind the user that *eye-tracking raw data MAY be stored under the `/sourcedata` directory*. IMHO, this is not acceptable for the specs as it is phrased at the moment.

Reply:
Yes, I agree it is the same spirit and I agree with the spirit. But the wording in the specs is very careful not to *push* or *encourage*. It only says that, *if someone knows they want to make original data available under /sourcedata, then actual data is preferred over links*.

This subtle difference is critical because BIDS cannot recommend anyone to share their original data. Encouraging that could push someone to leak private information or violate the license agreement of some experimental device.

In addition, if all there is to say about this is already in the specs, why risk here to create a conflict? IMHO, this paragraph doesn't belong in this BEP (nor any other, except the general rule were to be revisited).

On top of that and it might feel just a minor detail, but in the context of the other conversations about other potential (marked optional) columns, this paragraph felt to me like a way of saying "BIDS supports the minimum common denominator across ETs and if you want more go to the /sourcedata folder".

Reply:
The suggestion to store original data in `sourcedata` is also given for EEG 

> The data in their original format, if different from the supported formats, can be stored in the [/sourcedata directory](https://bids-specification.readthedocs.io/en/stable/common-principles.html#source-vs-raw-vs-derived-data).
> The original data format is especially valuable in case conversion elicits the loss of crucial metadata specific to manufacturers and specific EEG systems. 

and microscopy 

> If different from PNG, TIFF, OME-TIFF, or OME-ZARR, the original unprocessed data in the native format MAY be stored in the [/sourcedata directory](https://bids-specification.readthedocs.io/en/stable/common-principles.html#source-vs-raw-vs-derived-data).

and NIRS 

> Raw NIRS data in the native format, if different from SNIRF, can also be stored in the [/sourcedata](https://bids-specification.readthedocs.io/en/stable/common-principles.html#source-vs-raw-vs-derived-data) directory along with code to convert the data to SNIRF in the [/code](https://bids-specification.readthedocs.io/en/stable/common-principles.html#storage-of-derived-datasets) directory. The unprocessed raw data should be stored in the manufacturer's format before any additional processing or conversion is applied. Retaining the native file format is especially valuable in a case when conversion elicits the loss of crucial metadata unique to specific manufacturers and NIRS systems.

and possibly others.

Reply:
> The suggestion to store original data in `sourcedata` is also given for EEG

Dismissing the opinion that I'm convinced this item should not be "modality specific" (hence, only in the general sections of the spec such statements should be made), no example "encourages" releasing the data. They just inform that it can be done:

1. EEG: _[...] **can** be stored [...]_.
2. Microscopy: _[...] the original unprocessed data in the native format **MAY** be stored in [...]_
3. NIRS: _[...] **can** be stored [...]_.

Microscopy being the only one using the controlled language of **MAY**. All of them are incorrect (and against BIDS spirit) in adding the specifics that _if different from the supported formats_. Actually BIDS' spirit is sharing, so no discouragement to adding the original data to `/sourcedata` whatever the original format is (e.g., originally in `tsv.gz`), which is implicit in the clarification.

IMHO, this paragraph has no place as it reads right now. Considering these examples as a precedent, I could see a similar formulation (preferably the microscopy one). However, as it is written right now, it can be interpreted as a concealed way of saying _if you don't bother to fully convert the data (or don't know BIDS with sufficient depth to add and document with sidecars the new columns), we encourage you to add the original files so that someone else takes care of the problem_.

I don't like this suggestion, and would prefer complete removal of the sentence, but my reasoning could be summarized in the following rewriting:
```suggestion
As for all other modalities and datatypes supported by BIDS,
the original unprocessed data in the native format MAY be stored
in the [`/sourcedata` directory](https://bids-specification.readthedocs.io/en/stable/common-principles.html#source-vs-raw-vs-derived-data).
However, this possibility SHOULD NOT be employed as a means to enable minimal conversion
from the original format.
```

Because this suggestion should be placed in some modality-agnostic/common section (and perhaps articulate the SHOULD NOT part in a softer way), my opinion on this bit remains remove.

Reply:
Random thought.

How about we remove this phrase from this BEP and open a separate PR to also remove similar sentences from other data types and make sure it is expressed only once in the common principles in way we all agree on?

We have done similar "refactoring" in the past when we started noticing duplication / divergence across datatypes.

Reply:
> How about we remove this phrase from this BEP and open a separate PR to also remove similar sentences from other data types and make sure it is expressed only once in the common principles in way we all agree on?

I totally agree with this. Not only does it make the spec more consistent throughout, but it also makes it more readable because these boilerplates add unnecessary metadiscourse.

Reply:
To follow up - #1718